src/custom/src/OutboundEmailsDeployer/OutboundEmailsDeployer.php

```diff
@@ -314,38 +373,44 @@ protected function removeOrphanMailboxes($mapping)
                 }
 
                 if (!$found) {
-                    $emailAddress = \BeanFactory::getBean('EmailAddresses', $copy['email_address_id'], $beanOptions);
+                    $oe = \BeanFactory::retrieveBean('OutboundEmail', $copy['id'], $beanOptions);
 
-                    $user = \BeanFactory::getBean('Users', $copy['user_id'], $beanOptions);
+                    $user = $this->getUserNameFromId($copy['user_id']);
 
-                    $GLOBALS['log']->info(__METHOD__ . ' the user ' . $user->user_name . ' is not part of a team mapped to the outbound mailbox id ' . 
-                        $copy['parentmailbox_id']. ' for the email address ' . $emailAddress->email_address  . '. Deleting record id ' . $copy['id']);
+                    if (!empty($oe)) {
+                        $GLOBALS['log']->info(__METHOD__ . ' the user ' . $user['user_name'] . ' is not part of a team mapped to the outbound mailbox id ' .
+                            $copy['parentmailbox_id_c']. ' for the email address ' . $oe->email_address  . '. Deleting record id ' . $copy['id']);
 
-                    $oe = \BeanFactory::getBean('OutboundEmail', $copy['id'], $beanOptions);
-                    $oe->mark_deleted($copy['id']);
-   
-                    $message = sprintf(
-                        translate('LBL_OUTBOUND_EMAILS_DEPLOYER_MESSAGE_SUCCESSFUL_DELETE', 'Administration'),
-                        $emailAddress->email_address,
-                        $user->user_name
-                    );
+                        $oe->mark_deleted($copy['id']);
 
-                    $return['completed'][] = $message;
+                        $message = sprintf(
+                            translate('LBL_OUTBOUND_EMAILS_DEPLOYER_MESSAGE_SUCCESSFUL_DELETE', 'Administration'),
+                            $oe->email_address,
+                            $user['user_name']
+                        );
+
+                        $return['completed'][] = $message;
+                    }
                 }
             }
         }
         
         return $return;
     }
 
-    protected function areOutboundMailboxesInSync($originalMailbox, $usersMailbox)
+    protected function areOutboundMailboxesInSync($originalMailbox, $usersMailboxDB)
     {
         $this->enforcePermissions();
 
-        if (!empty($originalMailbox->id) && !empty($usersMailbox->id)) {
+        if (!empty($originalMailbox->id) && !empty($usersMailboxDB->id)) {
+            // decode the pass for comparison, since we now compare against the database values
+            foreach ($this->fieldsToDecode as $fieldToDecode) {
+                $usersMailboxDB->$fieldToDecode = Blowfish::decode(Blowfish::getKey($originalMailbox->module_key), $usersMailboxDB->$fieldToDecode);
```

Comment:
`SugarBean` has this function...

```
public function decrypt_after_retrieve($value)
{
    if(empty($value)) return $value; // no need to decrypt empty
    return Blowfish::decode($this->getEncryptKey(), $value);
}
```

You just need to make populate a bean with the data from the row you retrieved, if `$usersMailboxDB` isn't already a bean. But I wouldn't use the method directly, unless you have to. `SugarBean` also has...

```
public function &__get($varname)
{
    if(isset($this->encfields[$varname])) {
        $this->$varname = $this->decrypt_after_retrieve($this->encfields[$varname]);
        return $this->$varname;
    }
    $var = null;
    return $var;
}
```

Simply by making `$usersMailboxDB` a bean, you will automatically get the decoded value of the field anytime you use the getter via `$usersMailboxDB->mail_smtppass`. You can convert an array from a database row into a bean with...

```
$row = [...];
$usersMailbox = BeanFactory::newBean('OutboundEmail');
$usersMailbox->populateFromRow($row, true); // I'm not 100% whether or not you'll need to pass `true`. I would need to investigate it further. But that's what SugarBean::retrieve() does.
```

Basically, you can still fetch data using your own query, but you can instantiate a bean so that you get all of the conveniences that come with beans.